src/main/java/com/synopsys/integration/detect/lifecycle/run/step/IntelligentModeStepRunner.java

```diff
@@ -159,9 +197,15 @@ private void pollForBomCompletion(BlackDuckRunData blackDuckRunData, ProjectVers
         operationRunner.waitForBomCompletion(blackDuckRunData, bomToSearchFor);    
     }
 
-    public void uploadBdio(BlackDuckRunData blackDuckRunData, BdioResult bdioResult, CodeLocationAccumulator codeLocationAccumulator, Long timeout) throws OperationException {
+    public void uploadBdio(BlackDuckRunData blackDuckRunData, BdioResult bdioResult, Set<String> scanIdsToWaitFor, CodeLocationAccumulator codeLocationAccumulator, Long timeout) throws OperationException {
         BdioUploadResult uploadResult = operationRunner.uploadBdioIntelligentPersistent(blackDuckRunData, bdioResult, timeout);
         uploadResult.getUploadOutput().ifPresent(codeLocationAccumulator::addWaitableCodeLocations);
+        
+        if (uploadResult.getUploadOutput().isPresent()) {
+            for (UploadOutput result : uploadResult.getUploadOutput().get().getOutput()) {
+                result.getScanId().ifPresent((scanId) -> scanIdsToWaitFor.add(scanId));
```

Comment:
What should happen if the scan ID is not present?

Reply:
I haven't seen a case where the scanID doesn't come back for supported scans (package manager or signature scans). If for some reason we do get a valid result in everything but a scanID then we'd just not be able to get the scanId and we wouldn't be able to wait for it. I was thinking we just ignore it as there might be other scans to wait for vs erroring out. Thoughts?

Reply:
Yes, that works.